src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallRecords/RegistryWorkloadInstallationRecordRepository.cs

```diff
@@ -83,8 +85,44 @@ public IEnumerable<WorkloadId> GetInstalledWorkloads(SdkFeatureBand sdkFeatureBa
         {
             using RegistryKey wrk = _baseKey.OpenSubKey(Path.Combine(BasePath, $"{sdkFeatureBand}"));
 
+            return GetWorkloadInstallationRecordsFromRegistry(wrk);
+        }
+
+        public IEnumerable<WorkloadId> GetInstalledWorkloadsForFeatureBandAndBelow(SdkFeatureBand sdkFeatureBand)
+        {
+            using RegistryKey wrk = _baseKey.OpenSubKey(Path.Combine(BasePath, $"{sdkFeatureBand}"));
+            List<WorkloadId> allWorkloads = new();
+            var sdkBands = wrk?.GetSubKeyNames();
+
+            foreach (var band in sdkBands)
+            {
+                Log?.LogMessage($"Clean --all: Obtaining workload installation records for {BasePath} + {band}.");
+                try
+                {
+                    if (new SdkFeatureBand(band).CompareTo(sdkFeatureBand) < 1)
+                    {
+                        var sdkPath = Path.Combine(BasePath, band);
+                        allWorkloads.AddRange(GetWorkloadInstallationRecordsFromRegistry(_baseKey.OpenSubKey(sdkPath)));
+                    }
+                }
+                catch (Exception ex) when (
```

Comment:
Do we really only care about these exception types? Seems like any exception raised when trying to convert the version numbers should make us skip the invalid version, log it and move to the next one.

Reply:
So we ended up never using this function, even though I tested it. We can keep it as the work has already been done and it might be useful at some point, or just entirely get rid of it.
Thoughts?

#

If we do keep it:
We could catch the following: (But the registry access would've already caused a failure elsewhere, most likely.)
[ObjectDisposedException](https://learn.microsoft.com/en-us/dotnet/api/system.objectdisposedexception?view=net-7.0)
or 
[SecurityException](https://learn.microsoft.com/en-us/dotnet/api/system.security.securityexception?view=net-7.0)
can be thrown by the `OpenSubKey` but the registry validation would've already occured. 

We SHOULD catch the following: (can occur if the feature band contains a character that's an invalid path.)
[ArgumentException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentexception?view=net-7.0)

Reply:
Well. https://softwareengineering.stackexchange.com/questions/103031/should-i-remove-unreferenced-code The answer is pretty clear. Let's just remove this.